nifi/nifi-nar-bundles/nifi-flume-bundle/nifi-flume-processors/src/main/java/org/apache/nifi/processors/flume/NifiChannel.java

```diff
@@ -0,0 +1,31 @@
+
+package org.apache.nifi.processors.flume;
+
+import org.apache.flume.Context;
+import org.apache.flume.channel.BasicChannelSemantics;
+import org.apache.flume.channel.BasicTransactionSemantics;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+
+
+public class NifiChannel extends BasicChannelSemantics {
+  private final ProcessSession session;
+  private final Relationship relationship;
+
+  public NifiChannel(ProcessSession session, Relationship relationship) {
+    this.session = session;
+    this.relationship = relationship;
+  }
+
+  @Override
+  protected BasicTransactionSemantics createTransaction() {
+    return new NifiTransaction(session, relationship);
+  }
+
+  @Override
+  public void configure(Context context) {
+    throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```

Comment:
please leave out the autogenerated comment.

does this need to throw? could it just be a no-op?